server.go

```diff
@@ -66,12 +66,18 @@ func Accept(w http.ResponseWriter, r *http.Request) (*Conn, error) {
 type flushWrite struct {
 	w io.Writer
 	f http.Flusher
+	r *http.Request
```

Comment:
In the flushWriter you intend to use only the context, so I think it is better to store the context object and not the whole reqeuest.